### PR TITLE
Use netstandard friendly HttpMethod mechanisms

### PR DESCRIPTION
**Pull request by jzbrooks**
Addresses #19

### CODE REVIEW

**Comment by jzbrooks** [comment]
Seems like the check that failed is a build configuration issue. The build server can't download cake.

**Comment by ddunkin** [comment]
#21 updates Cake and builds on Ubuntu. That build error can be disregarded for this PR.

**Comment by ddunkin** [comment]
Reverting to using the static properties when available would save an allocation, yeah?
https://github.com/FacilityApi/FacilityCSharp/blob/e9813477fd92e9f502307d81a8466d8114b83901/src/Facility.CodeGen.CSharp/CSharpGenerator.cs#L478-L488

File: src/Facility.CodeGen.CSharp/CSharpGenerator.cs [hunk 1]
```diff
@@ -444,8 +445,22 @@ private CodeGenFile GenerateHttpMapping(HttpServiceInfo httpServiceInfo, Context
 								code.WriteLine($"new HttpMethodMapping<{requestTypeName}, {responseTypeName}>.Builder");
 								using (code.Block("{", "}.Build();"))
 								{
-									string httpMethodName = httpMethodInfo.Method.ToString().ToUpperInvariant();
-									code.WriteLine($"HttpMethod = new HttpMethod(\"{httpMethodName}\"),");
+									string httpMethod = httpMethodInfo.Method;
+									if (httpMethod == HttpMethod.Get.Method    ||
+									    httpMethod == HttpMethod.Put.Method    ||
```
Comment (ejball) [inline src/Facility.CodeGen.CSharp/CSharpGenerator.cs]: Nit, but please don't align operators (or anything except indentation).

File: src/Facility.CodeGen.CSharp/CSharpGenerator.cs [hunk 2]
```diff
@@ -444,8 +445,22 @@ private CodeGenFile GenerateHttpMapping(HttpServiceInfo httpServiceInfo, Context
...
-									string httpMethodName = CodeGenUtility.Capitalize(httpMethodInfo.Method.ToString().ToLowerInvariant());
-									code.WriteLine($"HttpMethod = HttpMethod.{httpMethodName},");
+									string httpMethod = httpMethodInfo.Method;
+									if (httpMethod == HttpMethod.Get.Method    ||
+									    httpMethod == HttpMethod.Put.Method    ||
+									    httpMethod == HttpMethod.Head.Method   ||
+									    httpMethod == HttpMethod.Post.Method   ||
+									    httpMethod == HttpMethod.Trace.Method  ||
+									    httpMethod == HttpMethod.Delete.Method ||
+									    httpMethod == HttpMethod.Options.Method)
```
Comment (ejball) [inline src/Facility.CodeGen.CSharp/CSharpGenerator.cs]: HEAD, TRACE, and OPTIONS are [not supported](https://github.com/FacilityApi/Facility/blob/b390fcccaf95017d37fb1bacf9d902d1a67426bb/src/Facility.Definition/Http/HttpMethodInfo.cs#L375) so we don't need to check for them here.

**Comment by ejball** [comment]
Fixed: 66705c1b86b3573ad7ec75027450bdb07a92ccda

**Pull request closed without merging**